dropshot/examples/self-referential.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Oxide Computer Company
+
+//! An example which demonstrates a server that can manage its own lifecycle.
+
+use dropshot::endpoint;
+use dropshot::ApiDescription;
+use dropshot::ConfigDropshot;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::HttpError;
+use dropshot::HttpResponseOk;
+use dropshot::HttpServerStarter;
+use dropshot::RequestContext;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+    let config_dropshot: ConfigDropshot = Default::default();
+    let config_logging =
+        ConfigLogging::StderrTerminal { level: ConfigLoggingLevel::Info };
+    let log = config_logging
+        .to_logger("example-self-referential")
+        .map_err(|error| format!("failed to create logger: {}", error))?;
+
+    let mut api = ApiDescription::new();
+    api.register(example_api_get_counter).unwrap();
+
+    let api_context = Arc::new(ExampleContext::new());
+
+    // Set up the server.
+    //
+    // Note that we split the "closer" from the "server", so we can continue
```

Comment:
This comment needs an update for the latest changes.

Reply:
Done